students/paul_c/session04/mailroom.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/local/bin/python3
+
+import sys
+from datetime import datetime
+
+donors = {
+    "Anne Ant": [1.00],
+    "Bonnie Bug": [20.00, 10.00],
+    "Chuck Cat": [2.00, 2.00, 2.00],
+    "Donna Dog": [5000.00, 2500.00, 1.50],
+    "Edna Ent": [.30, .50, .10]
+}
+
+email = {
+    "greeting": "\nHello {}\n\n",
+    "body": "We would like to thank you for your generous donation of ${}.\n\n",
+    "closing": "Best Regards,\n",
+    "signature": "The Foundation\n\n"
+}
+
+
+def send_all():
+    """
+    Function to send a thank you letter to all donors.
+    This function sums total donations and writes letters to timestamped files.
+    """
+    for key, value in donors.items():
+       # I could not figure out how to get the timestamp to work with the filename for open(). I kept getting a file does not exist error.
+       # timestamp = datetime.now().strftime('%D')
+       # filename = "{}.txt.".format(key) + str(timestamp)
+        filename = "{}.txt".format(key)
+        file_content = '{greeting}' '{body}' '{closing}' '{signature}'.format(**email).format(key, sum(value))
+        files = open(filename, "w")
+        files.write(file_content)
+        print(str(files) + "  ---File created.\n")
```

Comment:
make sure you call `files.close()` here if you aren't going to use the `with` command

Reply:
Added. Thanks!